scripts/tuple_tree_generator/tuple_tree_generator/templates/model_prelude.ts

```diff
@@ -0,0 +1,231 @@
+            }
+        }
+        return `${cleanStr(this.address, false)}:${this.type}` +
+               `${cleanStr(this.epoch)}${cleanStr(this.addressSpace)}`;
+    }
+}
+
+export type IReference = string;
+
+export class Reference {
+    reference: string;
+
+    constructor(ref: string) {
+        this.reference = ref;
+    }
+
+    toJSON(): string | undefined {
+        return this.reference;
+    }
+
+    static parse(ref?: string): Reference | undefined {
+        if(ref !== undefined) {
+            return new Reference(ref);
+        } else {
+            return undefined;
+        }
+    }
+}
+
+function randomBytes(length: number): string {
+    let value = "";
+    for (let i = 0; i < length; i++) {
+        value += (Math.random() * 16).toString(16);
+    }
+    return value;
+}
+
+function randomBigint(start: bigint, end: bigint): bigint {
+    const diff = end - start;
+    for (;;) {
+        const randomLength = diff.toString().length * Math.random();
+        const value = BigInt(`0x${randomBytes(randomLength)}`);
+        if (value < diff) {
+            return value + start;
+        }
+    }
+}
+
+export function genGuid(): string {
+    return randomBigint(2n ** 10n + 1n, 2n ** 64n - 1n).toString();
+}
+
+export function genPrimitiveTypeGuid(rawObject: IType): string {
+    const realObject = rawObject as IPrimitiveType;
+    const index = PrimitiveTypeKindValues.findIndex((e) => e === realObject.PrimitiveKind);
+    return BigInt(BigInt(index << 8) | realObject.Size).toString();
+}
+
+const genEnumTypeGuid = genGuid;
+const genTypedefTypeGuid = genGuid;
+const genStructTypeGuid = genGuid;
+const genUnionTypeGuid = genGuid;
+const genCABIFunctionTypeGuid = genGuid;
+const genRawFunctionTypeGuid = genGuid;
+
+try {
+    const yaml = require('yaml');
+
+    var _parseIModel: (file: string) => IModel = function(file: string) {
+        return yaml.parse(file, {
+            strict: false,
+            logLevel: "silent",
+            intAsBigInt: true,
+        });
+    }
+
+    var _parseModel: (file: string) => Model = function(file: string) {
+        return new Model(parseIModel(file));
+    }
+
+    var _dumpModel: (model: IModel | Model) => string = function(model: Model | IModel) {
+        const parseOptions = {
+        keepUndefined: false,
+        sortMapEntries: true,
+        } as const;
+
+        const options = {
+        doubleQuotedAsJSON: false,
+        defaultKeyType: yaml.Scalar.PLAIN,
+        defaultStringType: yaml.Scalar.QUOTE_DOUBLE,
+        directives: true
+        } as const;
+
+        const doc = new yaml.Document(model, parseOptions);
+        return doc.toString(options);
+    }
+} catch(err) {
+    var _parseIModel = (undefined as (file: string) => IModel);
+    var _parseModel = (undefined as (file: string) => Model);
+    var _dumpModel = (undefined as (model: Model | IModel) => string);
+    console.log("revng-model: package yaml not found, not creating parse wrappers")
+}
```

Comment:
This `try ... catch` seems too big.

Reply:
removed the `try ... catch`, now the generated file is a bit more opinionated.